File: 9.回文数.py
#
# @lc app=leetcode.cn id=9 lang=python3
#
# [9] 回文数
#

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        cur=0
        if x < 0 or x%10==0 and x!=0:
            return False
        else:
            num = x
            while num!=0:
                cur=cur*10 +num%10
                num//=10   
            return cur==x
# @lc code=end

# Solution1：
# class Solution:
#     def isPalindrome(self, x: int) -> bool:
#         if x < 0:
#             return False
#         elif x == 0:
#             return True
#         else:
#             string = str(x)
#             lenth = len(string)
#             for i in range(lenth // 2):
#                 if string[i] != string[lenth - i - 1]:
#                     return False
#             return True


# Solution2:
# class Solution:
#     def isPalindrome(self, x: int) -> bool:
#         return str(x)==str(x)[::-1]

# Solution3:
# class Solution:
#     def isPalindrome(self, x: int) -> bool:
#         cur=0
#         if x < 0 or x%10==0 and x!=0:
#             return False
#         else:
#             num = x
#             while num!=0:
#                 cur=cur*10 +num%10
#                 num//=10   
#             return cur==x

File: 9.回文数.java
import java.util.ArrayList;
import java.util.List;
/*
 * @lc app=leetcode.cn id=9 lang=java
 *
 * [9] 回文数
 */

// @lc code=start
class Solution {
    public boolean isPalindrome(int x) {
        if(x==0){
            return true;
        }else if(x<0||x%10==0){
            return false;
        }
        int reverseNum=0;
        while(reverseNum<x){
            reverseNum=reverseNum*10+x%10;
            x/=10;
        }
        return reverseNum==x||reverseNum/10==x;
        
    }
}
// @lc code=end


// Solution1:
// class Solution {
//     public boolean isPalindrome(int x) {
//         if(x==0){
//             return true;
//         }else if(x<0){
//             return false;
//         }
//         StringBuilder sBuilder=new StringBuilder(String.valueOf(x));
//         if(sBuilder.toString().equals(new StringBuilder(sBuilder).reverse().toString())){
//             return true;
//         }else{
//             return false;
//         }

//     }
// }

// Solution2:
// class Solution {
//     public boolean isPalindrome(int x) {
//         if(x==0){
//             return true;
//         }else if(x<0){
//             return false;
//         }
//         List<Integer> list=new ArrayList<>();
//         while(x!=0){
//             list.add(x%10);
//             x/=10;
//         }
//         int len=list.size();
//         for(int i=0;i<len;i++){
//             if(list.get(i)!=list.get(len-i-1)){
//                 return false;
//             }
//         }
//         return true;
//     }
// }

// Solution3:
// class Solution {
//     public boolean isPalindrome(int x) {
//         if(x==0){
//             return true;
//         }else if(x<0||x%10==0){
//             return false;
//         }
//         int reverseNum=0;
//         while(reverseNum<x){
//             reverseNum=reverseNum*10+x%10;
//             x/=10;
//         }
//         return reverseNum==x||reverseNum/10==x;
        
//     }
// }